src/onyx/log/commands/common.clj

```diff
@@ -96,7 +96,8 @@
         active? (partial at-least-one-active? replica)]
     (every? identity (map #(active? (get-in replica [:allocations job %])) tasks))))
 
-(defn transform-job-allocations [peer-state allocations]
+(defn job-receivable-peers [peer-state allocations job-id]
+  ;; can also filter down tasks?
```

Comment:
I'll improve it, it's mostly a TODO. What it means is that we know that a peer on a particular task should filter receivable peers further, as only peers on egress tasks are required.